app/controllers/subscriptions_controller.rb

```diff
@@ -2,21 +2,24 @@ class SubscriptionsController < ApplicationController
   def create
     return render json: { id: 0 }, status: :created if smoke_test_address?
 
-    subscription = Subscription.find_or_initialize_by(
+    existing_subscription = Subscription.active.find_by(
```

Comment:
I wonder if you might want this in a lock for updates to prevent race conditions.

Reply:
Agreed, I've gone with a transaction though. Locking would be annoying with all the various different objects we've got, but a transaction will catch race conditions as the unique index on active subscriptions will only allow one transaction to succeed.

Reply:
It wouldn't actually be too tricky. You'd need to start a transaction and then add a lock to the Subscription find_by method chain.

The advantage you get over that than just DB locking is that it'll return the appropriate HTTP status code, whereas db locking you'll get a 500 (unless you go down the catching route)

Reply:
Oh yeah, I forgot about the appropriate HTTP status codes... I figured we might have to lock the subscriber as well, but since we're in a transaction we don't need to. Amended the PR with the `lock` call!